content/docs/reference/config/builder-config.md

```diff
@@ -73,6 +73,15 @@ A [builder][builder] configuration schema is as follows:
     > at the time of `pack`'s release. In other words, for a particular version of `pack`, this default
     > will not change despite new lifecycle versions being released.
 
+- #### `build.env` _(optional)_
+  The [[build.env]] is used to override platform environment variables at build-time. use Env `CNB_BUILD_CONFIG_DIR` to override when building image from cli to change default(`/cnb/build-config`) directory of these build-time environment variables stored
+
+  - **`name`** _(string, required)_\
+    The Name/Key of the Environment Variable. If Platform Environment Variable key/name is specified, then it is overridden at build-time else it will create a build-time environment variable with the given `name` that is not defined(key/name doesn't exists) at runtime of the container
+
+  - **`value`** _(string, required)_\
```

Comment:
Should we mention suffix and delim here? We can link to https://github.com/buildpacks/spec/blob/main/buildpack.md#environment-variable-modification-rules to make the text here more terse if we want to.